public/src/views/CalendarView.jsx

```diff
@@ -1,10 +1,108 @@
-import { React } from 'react';
-import { PropTypes } from 'prop-types';
+import { React, useEffect, useState } from 'react';
+// import { PropTypes } from 'prop-types';
+import styled, { css } from 'styled-components';
+import { ComposableInput, StyledInput } from '../components/ComposableInput';
+import ButtonGroup from '../components/ButtonGroup';
+import CheckboxButton from '../components/CheckboxButton';
+import API from '../hooks';
+import { formatDate } from '../utils';
+import Day, { DaysContainer } from '../components/Day';
+
+const CalendarContainer = styled.div`
+  /* ${({ theme }) => css`
+    max-width: ${theme.maxWidth};
+  `} */
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  margin: 1em auto;
+`;
+const CalendarFormContainer = styled.div`
+  ${({ theme }) => css`
+    width: ${theme.maxWidth};
+  `}
+`;
 
 export const CalendarView = () => {
+  const [day, setDayOrMonth] = useState(true);
+  const [data, setData] = useState([]);
+  const [shownDate, setShownDate] = useState(new Date());
+  const retrieveDate = async () => {
+    const date = formatDate(shownDate).substr(0, 10);
+    const response = await API.retrieveEventsForDay(date);
+    setData([response]);
+  };
+  const handleDateTimeInput = (e) => {
+    if (!e.target.validity.valid) return;
+    const [date, time] = e.target.value.split('T');
+    const [year, month, d] = date.split('-');
+    const [hour, minutes] = time.split(':');
+    const dt = new Date(Number(year), Number(month) - 1, Number(d), Number(hour), Number(minutes));
+    setShownDate(dt);
+  };
+  const retrieveWeek = async () => {
+    const END_OF_WEEK = 6;
+    const startOfWeek = shownDate.getDay() === 0
+      ? shownDate
+      : (() => { const a = new Date(); a.setDate(shownDate.getDate() - shownDate.getDay()); return a; })();
+    const endOfWeek = shownDate.getDay() === 6
+      ? shownDate
+      : (() => {
+        const a = new Date(); a.setDate(shownDate.getDate() + (END_OF_WEEK - shownDate.getDay())); return a;
+      })();
+    const response = await API.retrieveEventsForDuration(
+      formatDate(startOfWeek).substr(0, 10),
+      formatDate(endOfWeek).substr(0, 10),
+    );
+    const groupByDate = (ajaxData) => {
+      const groupedObj = ajaxData.reduce((acc, cur) => {
+        const [date, time] = cur.datetime.split('T');
+        Array.isArray(acc[date]) ? acc[date].push(cur) : acc[date] = [cur];
```

Comment:
This has terrible readability

You're doing two statements, not an assignment. if/else bruh